src/amo/reducers/collections.js

```diff
@@ -449,21 +451,78 @@ export const addAddonToCollection = ({
   };
 };
 
-type UpdateCollectionParams = {|
+export type RequiredModifyCollectionParams = {|
   errorHandlerId: string,
-  collectionSlug: string,
+  user: string,
+|};
+
+export type OptionalModifyCollectionParams = {|
   defaultLocale: ?string,
   description: ?LocalizedString,
+|};
+
+type CreateCollectionParams = {|
+  ...RequiredModifyCollectionParams,
+  ...OptionalModifyCollectionParams,
+  name: LocalizedString,
+  slug: string,
+|};
+
+type UpdateCollectionParams = {|
+  ...RequiredModifyCollectionParams,
+  ...OptionalModifyCollectionParams,
+  collectionSlug: string,
   name: ?LocalizedString,
   slug: ?string,
-  user: string,
+|};
+
+export type CreateCollectionAction = {|
+  type: typeof CREATE_COLLECTION,
+  payload: CreateCollectionParams,
 |};
 
 export type UpdateCollectionAction = {|
   type: typeof UPDATE_COLLECTION,
   payload: UpdateCollectionParams,
 |};
 
+export const validateRequiredCollectionParams = ({
+  errorHandlerId,
+  user,
+}: RequiredModifyCollectionParams) => {
+  invariant(errorHandlerId, 'errorHandlerId is required');
+  invariant(user, 'user is required');
+};
+
+export const createCollection = ({
+  errorHandlerId,
+  defaultLocale,
+  description,
+  name,
+  slug,
+  user,
+}: CreateCollectionParams = {}): CreateCollectionAction => {
+  validateRequiredCollectionParams({
```

Comment:
This is actually more lines than just using the two invariant lines:

```js
invariant(errorHandlerId, 'errorHandlerId is required');
invariant(user, 'user is required');
```

I get it doesn't duplicate the messages but we already dupe them elsewhere and invariant should let us nix all the messages in production anyway. I'd rather us just use invariant directly.